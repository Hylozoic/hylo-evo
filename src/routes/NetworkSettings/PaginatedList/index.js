import React, { Component } from 'react'

import cx from 'classnames'
import { times, isEmpty } from 'lodash/fp'
import RemovableListItem from 'components/RemovableListItem'
import { communityUrl } from 'utils/index'

import '../NetworkSettings.scss'

export default class PaginatedList extends Component {
  constructor (props) {
    super(props)
    this.state = {
      prevItems: []
    }
  }

  updatePrevItems () {
    const { pending, items } = this.props
    const { prevItems } = this.state
    if (pending || items === prevItems || isEmpty(items)) return
    this.setState({prevItems: items})
  }

  componentDidMount () {
    this.updatePrevItems()
  }

  componentDidUpdate () {
    this.updatePrevItems()
  }

  render () {
    const {
      className,
      items,
      itemProps,
      label,
      page,
      pageCount,
      pending,
      removeItem,
      setPage
    } = this.props
    const { prevItems } = this.state
    const visibleItems = pending ? prevItems : items

    return <div styleName={cx('paginated-list', {loading: pending})} className={className}>
      <div styleName='section-label'>{label}</div>
      {visibleItems.map(m => <RemovableListItem
        item={m}
        url={communityUrl(m.slug)}
        key={m.id}
        removeItem={removeItem}
        {...itemProps} />)}
      <PaginationLinks page={page} pageCount={pageCount} setPage={setPage} />
    </div>
  }
}

export function PaginationLinks ({ pageCount, setPage, page }) {
  if (pageCount < 2) return null

  function PageLink ({ i }) {
    const current = i === page
    return <span styleName={current ? 'page-current' : 'page-link'} onClick={() => !current && setPage(i)}>{i + 1}</span>
  }

  return <div styleName='pagination-links'>
    {times(i => <PageLink key={i} i={i} />, pageCount)}
  </div>
}
