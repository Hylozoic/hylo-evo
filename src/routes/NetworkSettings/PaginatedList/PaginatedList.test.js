import PaginatedList, { PaginationLinks } from './index'
import { shallow } from 'enzyme'
import React from 'react'

describe('PaginatedList', () => {
  it('renders correctly', () => {
    const items = [{id: 2}, {id: 3}]

    const wrapper = shallow(<PaginatedList
      isAdmin
      items={items}
      page={3}
      pageCount={5}
      setPage={() => {}}
      itemProps={{square: true, size: 40}} />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe('PaginationLinks', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<PaginationLinks
      isAdmin
      page={3}
      pageCount={5}
      setPage={() => {}} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders null with 1 page', () => {
    const wrapper = shallow(<PaginationLinks
      page={1}
      pageCount={1}
      setPage={() => {}} />)
    expect(wrapper.html()).toEqual(null)
  })
})
