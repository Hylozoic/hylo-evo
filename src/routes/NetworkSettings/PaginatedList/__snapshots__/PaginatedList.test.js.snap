// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginatedList renders correctly 1`] = `
<div
  data-stylename="paginated-list"
>
  <div
    data-stylename="section-label"
  />
  <RemovableListItem
    item={
      Object {
        "id": 2,
      }
    }
    key="2"
    size={40}
    square={true}
    url="/all"
  />
  <RemovableListItem
    item={
      Object {
        "id": 3,
      }
    }
    key="3"
    size={40}
    square={true}
    url="/all"
  />
  <PaginationLinks
    page={3}
    pageCount={5}
    setPage={[Function]}
  />
</div>
`;

exports[`PaginationLinks renders correctly 1`] = `
<div
  data-stylename="pagination-links"
>
  <PageLink
    i={0}
    key="0"
  />
  <PageLink
    i={1}
    key="1"
  />
  <PageLink
    i={2}
    key="2"
  />
  <PageLink
    i={3}
    key="3"
  />
  <PageLink
    i={4}
    key="4"
  />
</div>
`;
